### ISSUE TITLE
About the number of instances

### ISSUE THREAD

**Issue by Whoo-jl** [junjiehe96/FastInst#2]
There are only six categories, but the inference result shows that a total of 100 instances are detected, what is the reason?

**Comment by junjiehe96** [comment]
We output `test_topk_per_image=100` instances by default for coco evaluation. In actual use, you can try to set a classification threshold, e.g. 0.5, to further filter the output.

**Comment by Whoo-jl** [comment]
![image](https://user-images.githubusercontent.com/52949064/231973787-cf292292-f003-406e-98cc-6153cf9d330b.png)
Does the classification threshold refer to the value of 0.5 in the picture?

**Comment by Whoo-jl** [comment]
Do I need to set 'NUM_OBJECT_QUERIES' and 'DETECTIONS_PER_IMAGE' to my number of categories 6 as well?

**Comment by junjiehe96** [comment]
> ![image](https://user-images.githubusercontent.com/52949064/231973787-cf292292-f003-406e-98cc-6153cf9d330b.png) Does the classification threshold refer to the value of 0.5 in the picture? I set it to 0.9 and it still outputs 100 instances

No, this is the threshold for mask predictions. The classification threshold is set for `result.scores`.

**Comment by junjiehe96** [comment]
> Do I need to set 'NUM_OBJECT_QUERIES' and 'DETECTIONS_PER_IMAGE' to my number of categories 6 as well?

If there are only a few instances in an image, you can try reducing "NUM_OBJECT_QUERIES" for better speed-performance trade-off. But this may require you to retrain the model.

'DETECTIONS_PER_IMAGE' generally does not need to be changed. It is used for evaluations.

**Comment by Whoo-jl** [comment]
> > ![image](https://user-images.githubusercontent.com/52949064/231973787-cf292292-f003-406e-98cc-6153cf9d330b.png) Does the classification threshold refer to the value of 0.5 in the picture? I set it to 0.9 and it still outputs 100 instances
> 
> No, this is the threshold for mask predictions. The classification threshold is set for `result.scores`.

> > Do I need to set 'NUM_OBJECT_QUERIES' and 'DETECTIONS_PER_IMAGE' to my number of categories 6 as well?
> 
> If there are only a few instances in an image, you can try reducing "NUM_OBJECT_QUERIES" for better speed-performance trade-off. But this may require you to retrain the model.
> 
> 'DETECTIONS_PER_IMAGE' generally does not need to be changed. It is used for evaluations.

Thank you for your patience, under your guidance my problem has been successfully solved!😄

**Comment by cLEARLoVE00** [comment]
can you tell me how to slover this problem, I have meet this question? @ @Whoo-jl
